src/awkward/contents/regulararray.py

```diff
@@ -420,19 +417,25 @@ def _getitem_next(
 
         elif isinstance(head, slice):
             nexthead, nexttail = ak._slicing.headtail(tail)
-            start, stop, step = head.indices(self._size)
-
-            nextsize = 0
-            if step > 0 and stop - start > 0:
-                diff = stop - start
-                nextsize = diff // step
-                if diff % step != 0:
-                    nextsize += 1
-            elif step < 0 and stop - start < 0:
-                diff = start - stop
-                nextsize = diff // (step * -1)
-                if diff % step != 0:
-                    nextsize += 1
+            # Handle unknown sizes
+            if self._size is None:
+                nextsize = None
+                start = 0
+                step = 1 if head.step is None else head.step
```

Comment:
If `step` is negative, should `start` be `0`?

Slices are completely different when the `step` is negative; be sure there is a test case for it.